ai_baset: output now requires function identifier [blocks: #3126]

tautschnig opened the pull request:
We are working towards removing the "function" field from
goto_programt::instructionst::instructiont, and thus need to pass the identifier
of the function whenever it actually is required.

<!---
Thank you for your contribution. Please make sure your pull request fulfils all of the below requirements. If you cannot currently tick all the boxes, but would still like to create a PR, then add the label "work in progress" and assign the PR to yourself.
--->

- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- n/a The feature or user visible behaviour I have added or modified has been documented in the User Guide in doc/cprover-manual/
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

<!---
See, e.g., https://chris.beams.io/posts/git-commit/ for general guidelines on commit messages.

If you have created commits mixing multiple features and/or unrelated white-space changes, use a sequence involving git reset and git add -p to fix this.
--->

kroening committed:
ai_baset: output now requires function identifier

We are working towards removing the "function" field from
goto_programt::instructionst::instructiont, and thus need to pass the identifier
of the function whenever it actually is required.

smowton commented on src/analyses/ai.h:
```diff
@@ -124,6 +124,18 @@ class ai_baset
 
+  /// Output the abstract states for a single function
+  /// \param ns: The namespace
+  /// \param identifier: The identifier used to find a symbol to identify the
```
`to identify the...` (copy and paste error?)

tautschnig replied:
Not really, the next parameter `goto_program,` is what is being identified here. But that's a bit weird to read. Fixed.

smowton commented:
Are the parameter order changes just for consistency?

tautschnig committed:
Make order of arguments in ai_baset::output* consistent

Place the identifier (the name of the function) before the goto program.

tautschnig committed:
ai_baset::initialize now requires a function name

We are working towards removing the "function" field from
goto_programt::instructionst::instructiont, and thus need to pass the identifier
of the function whenever it actually is required.

tautschnig commented:
> Are the parameter order changes just for consistency?

Yes - I've now split this up in three commits.

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 500d4850ab819384070d59543e1e2271d097781d).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/97924789

peterschrammel commented on src/analyses/ai.cpp:
```diff
@@ -187,12 +187,14 @@ void ai_baset::entry_state(const goto_programt &goto_program)
 
-void ai_baset::initialize(const goto_functionst::goto_functiont &goto_function)
+void ai_baset::initialize(
+  const irep_idt &function,
```
While doing this could we agree on a consistent name, e.g. `function_id`, `function_identifier`, `function`, `function_name` or so?

tautschnig replied:
Done (now using `function_identifier`).

peterschrammel reviewed (approved):
LGTM modulo naming

tautschnig committed:
ai_baset: consistently use function_id instead of function_identifier

It is as clear, but makes for shorter lines.

allredj reviewed (commented):
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: d7db14f8ecf9f5e62166f77f1ebbf43aca0149ae).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/97965595
